src/Tools/Console/Command/DiffCommand.php

```diff
@@ -63,6 +63,12 @@ protected function configure(): void
                 InputOption::VALUE_NONE,
                 'Format the generated SQL.',
             )
+            ->addOption(
+                'nowdoc',
+                null,
+                InputOption::VALUE_NONE,
```

Comment:
How about providing `no-nowdoc` as well?

```suggestion
                InputOption::VALUE_NEGATABLE,
```

Reply:
Why not.
What about formatted output ? do you prefer to always keep it as nowdoc no mater this option or not?

Reply:
I'm not sure why exactly one would want to disable nowdoc in the first place. Can you elaborate? You mentioned that it's "sad" that it cannot be disabled, but why is that exactly? Depending on your answer, it might be easier to make a decision.

Reply:
I think `--formatted` should imply `--nowdoc`, but that if we're going to provide a means to disable nowdoc, it would be weird to ignore it.

Reply:
@greg0ire even with formatted I don't think imply `--nowdoc` is a great idea for all query
I would adopt the same way of working of `--formatted` which is limited by `--line-length`

I no formatting is applied, this will result in a one-line query that provides no value for nowdoc.
As a quick example as well, adding one "legacy_id" column to 25 tables will generate 25 short querry, i think it should stay on one line, avoiding to have 75 lines (times 2 for up and down) that don't bring anything 

To resume what I propose: 
- By default, no nowdoc
- If `--formatted` enabled, all formatted output will be by default in nowdoc (regarding `--line-length` option)
- if `--nowdoc` enabled, all output will be in nowdoc
- if `--no-nowdoc` enabled, all output will be as previously formatted without nowdoc, event formatted ones

If we agree on this behavior, I'll implement it like this.

Reply:
To clarify, what is the difference between formatted and nowdoc then?

Reply:
@theofidry
- default

```php
$this->addSql('CREATE TABLE "user" (id SERIAL PRIMARY KEY, firstname VARCHAR(100) NOT NULL, lastname VARCHAR(100) NOT NULL )');
$this->addSql('ALTER TABLE platform ADD test_number INT NOT NULL');
```
- `--nowdoc`
```php
  $this->addSql(<<<'SQL'
      CREATE TABLE "user" (id SERIAL PRIMARY KEY, firstname VARCHAR(100) NOT NULL, lastname VARCHAR(100) NOT NULL)
  SQL);
  $this->addSql(<<<'SQL'
      ALTER TABLE platform ADD test_number INT NOT NULL
  SQL);
  ```

- `--formated`
```php
$this->addSql(<<<'SQL'
  CREATE TABLE "user" (
    id SERIAL PRIMARY KEY,
    firstname VARCHAR(100) NOT NULL,
    lastname VARCHAR(100) NOT NULL
  )
SQL);
$this->addSql('ALTER TABLE platform ADD test_number INT NOT NULL');
```

- `--formated --nowdoc`
```php
$this->addSql(<<<'SQL'
  CREATE TABLE "user" (
    id SERIAL PRIMARY KEY,
    firstname VARCHAR(100) NOT NULL,
    lastname VARCHAR(100) NOT NULL
  )
SQL);
$this->addSql(<<<'SQL'
    ALTER TABLE platform ADD test_number INT NOT NULL
SQL);
```
- `--formated --no-nowdoc`
```php
$this->addSql('
  CREATE TABLE "user" (
    id SERIAL PRIMARY KEY,
    firstname VARCHAR(100) NOT NULL,
    lastname VARCHAR(100) NOT NULL
')
SQL);
$this->addSql('ALTER TABLE platform ADD test_number INT NOT NULL');
```

Reply:
I think I'm OK with this proposal.